build-scripts/package-tests.cake

```diff
@@ -0,0 +1,382 @@
+    {
+        Console.WriteLine("\n============================================="); ;
+        Console.WriteLine(message);
+        Console.WriteLine("=============================================");
+    }
+
+    private void DisplayTestEnvironment(PackageTest test)
+    {
+        Console.WriteLine("Test Environment");
+        Console.WriteLine($"   OS Version: {Environment.OSVersion.VersionString}");
+        Console.WriteLine($"  CLR Version: {Environment.Version}");
+        Console.WriteLine($"    Arguments: {test.Arguments}");
+        Console.WriteLine();
+    }
+}
+
+public abstract class NetFXPackageTester : PackageTester
+{
+    public NetFXPackageTester(BuildParameters parameters) : base(parameters)
+    {
+        // Level 1 tests are run each time we build the packages
+        PackageTests.Add(new PackageTest(1, "Run mock-assembly.dll under .NET 3.5",
+            "net35/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 37,
+                Passed = 23,
+                Failed = 5,
+                Warnings = 0,
+                Inconclusive = 1,
+                Skipped = 7
+            }));
+
+        PackageTests.Add(new PackageTest(1, "Run mock-assembly.dll under .NET 4.x",
+            "net40/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 37,
+                Passed = 23,
+                Failed = 5,
+                Warnings = 0,
+                Inconclusive = 1,
+                Skipped = 7
+            }));
+
+        PackageTests.Add(new PackageTest(1, "Run both copies of mock-assembly together",
+            "net35/mock-assembly.dll net40/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 2 * 37,
+                Passed = 2 * 23,
+                Failed = 2 * 5,
+                Warnings = 0,
+                Inconclusive = 2 * 1,
+                Skipped = 2 * 7
+            }));
+
+        PackageTests.Add(new PackageTest(1, "Run project with both copies of mock-assembly",
+            $"../../NetFXTests.nunit --config={_parameters.Configuration}",
+            new ExpectedResult("Failed")
+            {
+                Total = 2 * 37,
+                Passed = 2 * 23,
+                Failed = 2 * 5,
+                Warnings = 0,
+                Inconclusive = 2 * 1,
+                Skipped = 2 * 7
+            },
+            NUnitProjectLoader));
+    }
+
+    //// Level 2 tests are run for PRs and when packages will be published
+
+    //PackageTests.Add(new PackageTest(2, "Run mock-assembly.dll built for NUnit V2",
+    //    "v2-tests/mock-assembly.dll",
+    //    new ExpectedResult("Failed")
+    //    {
+    //        Total = 28,
+    //        Passed = 18,
+    //        Failed = 5,
+    //        Warnings = 0,
+    //        Inconclusive = 1,
+    //        Skipped = 4
+    //    },
+    //    NUnitV2Driver));
```

Comment:
As far as I can tell this is the only place where the new mock assembly is mentioned?

Reply:
Yes... the goal is to activate this test in a follow-up PR.

Reply:
I think this should be uncommented and in use, right?